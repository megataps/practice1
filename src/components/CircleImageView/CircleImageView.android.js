import React, { Component, PropTypes } from 'react';
import {
    View,
    Text,
    StyleSheet,
    Image
} from 'react-native';

import Colors from 'config/colors';

export default class CircleImageView extends Component {
    render() {
        return (
            <View style={styles.container}>
                <Image
                    style={styles.icon}
                    source={this.props.iconUrI} />
            </View>
        )
    }
}

CircleImageView.propTypes = {
  iconUrI: PropTypes.number.isRequired
};

const IMAGE_SIZE = 90;
const IMAGE_ICON_SIZE = IMAGE_SIZE / 2;

const styles = StyleSheet.create({
    container: {
        width: IMAGE_SIZE,
        height: IMAGE_SIZE,
        borderRadius: IMAGE_ICON_SIZE,
        backgroundColor: Colors.white,
        justifyContent: 'center',
        alignItems: 'center'
    },

    icon: {
        width: IMAGE_ICON_SIZE,
        height: IMAGE_ICON_SIZE
    }
});