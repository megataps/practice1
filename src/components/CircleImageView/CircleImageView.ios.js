import React, {Component} from 'react';
import {
    View,
    Text,
    StyleSheet
} from 'react-native';

export default class CircleImageView extends Component {
    render () {
        return (
           <View style={styles.container}>
                <Text>I'm the CircleImageView component IOS</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});