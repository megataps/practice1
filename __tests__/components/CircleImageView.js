import 'react-native';
import React from 'react';
import CircleImageView from 'components/CircleImageView/CircleImageView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <CircleImageView
      iconUrI={require('assets/images/check_red.png')}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
