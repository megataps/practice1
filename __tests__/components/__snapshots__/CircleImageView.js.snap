// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 45,
      "height": 90,
      "justifyContent": "center",
      "width": 90,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 45,
        "width": 45,
      }
    }
  />
</View>
`;
